quantum/nodes/grovers-algo/grovers-algo.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const util = require('util');
+const snippets = require('../../snippets');
+const shell = require('../../python').PythonShell;
+const errors = require('../../errors');
+
+module.exports = function(RED) {
+  function GroversAlgoNode(config) { // Change name
+    RED.nodes.createNode(this, config);
+    this.name = config.name || 'Grover\'s algorithm';
+    const node = this;
+
+    this.on('input', async function(msg, send, done) {
+      let error = errors.validateGroversInput(msg);
+      if (error) {
+        done(error);
+        return;
+      }
+      const script = util.format(snippets.GROVERS_ALGO, msg.payload);
+      await shell.restart();
```

Comment:
Should this `restart()` call be here? This will reset the entire Python script when run, and should usually only be placed at the very start of the circuit (e.g. in `quantum-circuit.js`).

Reply:
I believe this node will not be connected to circuit node, it is rather standalone node. If I didn't use restart command than I receive error:
```
(node:5551) UnhandledPromiseRejectionWarning: Error: Python process has not been started - call start() before executing commands.
```
Although, instead of restart we could use start command

Reply:
In that case the ideal thing to do is to give this node its own `PythonShell` instance, rather than using the global one. Then call `await shell.start()` at the very start of the input listener, and `shell.stop()` at the very end of the input listener.

Reply:
Thanks, sound good!